pkg/resource/prunercronjob.go

```diff
@@ -0,0 +1,237 @@
+	batchlisters "k8s.io/client-go/listers/batch/v1beta1"
+
+	imageregistryapiv1 "github.com/openshift/api/imageregistry/v1"
+	imageregistryv1listers "github.com/openshift/client-go/imageregistry/listers/imageregistry/v1"
+	"github.com/openshift/cluster-image-registry-operator/defaults"
+	"github.com/openshift/cluster-image-registry-operator/pkg/parameters"
+)
+
+const (
+	cronJobOwnerAnnotation = "imageregistry.openshift.io"
+)
+
+var (
+	defaultSuspend                          = false
+	defaultSchedule                         = "*/1 * * * *"
+	defaultStartingDeadlineSeconds    int64 = 60
+	defaultFailedJobsHistoryLimit     int32 = 3
+	defaultSuccessfulJobsHistoryLimit int32 = 3
+	defaultKeepTagRevisions                 = 3
+	defaultKeepYoungerThan                  = "60m"
+	defaultTolerations                      = []kcorev1.Toleration{}
+	defaultNodeSelector                     = map[string]string{}
+	defaultResources                        = kcorev1.ResourceRequirements{
+		Requests: kcorev1.ResourceList{
+			kcorev1.ResourceCPU:    resource.MustParse("100m"),
+			kcorev1.ResourceMemory: resource.MustParse("256Mi"),
+		},
+	}
+	defaultAffinity kcorev1.Affinity
+)
+
+// PrunerCronJobIsCreatedByOperator returns whether or not the PrunerCronJob was created by the Operator
+func PrunerCronJobIsCreatedByOperator(prunerCronJob *batchapi.CronJob) bool {
+	_, ok := prunerCronJob.Annotations[cronJobOwnerAnnotation]
+	return ok
+}
+
+var _ Mutator = &generatorPrunerCronJob{}
+
+type generatorPrunerCronJob struct {
+	lister       batchlisters.CronJobNamespaceLister
+	client       batchset.BatchV1beta1Interface
+	prunerLister imageregistryv1listers.ImagePrunerLister
+	cr           *imageregistryapiv1.ImagePruner
+}
+
+func newGeneratorPrunerCronJob(lister batchlisters.CronJobNamespaceLister, client batchset.BatchV1beta1Interface, prunerLister imageregistryv1listers.ImagePrunerLister, params *parameters.Globals) *generatorPrunerCronJob {
+	return &generatorPrunerCronJob{
+		lister:       lister,
+		client:       client,
+		prunerLister: prunerLister,
+	}
+}
+
+func (gcj *generatorPrunerCronJob) Type() runtime.Object {
+	return &batchapi.CronJob{}
+}
+
+func (gcj *generatorPrunerCronJob) GetGroup() string {
+	return batchapi.GroupName
+}
+
+func (gcj *generatorPrunerCronJob) GetResource() string {
+	return "batches"
+}
+
+func (gcj *generatorPrunerCronJob) GetNamespace() string {
+	return defaults.ImageRegistryOperatorNamespace
+}
+
+func (gcj *generatorPrunerCronJob) GetName() string {
+	return "image-pruner"
+}
+
+func (gcj *generatorPrunerCronJob) expected() (runtime.Object, error) {
+	cr, err := gcj.prunerLister.Get(defaults.ImageRegistryImagePrunerResourceName)
+	if err != nil {
+		return nil, err
+	}
+	gcj.cr = cr
+
+	cj := &batchapi.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        gcj.GetName(),
+			Namespace:   gcj.GetNamespace(),
+			Annotations: map[string]string{cronJobOwnerAnnotation: "true"},
```

Comment:
Why do you need this annotation?

Reply:
to show that the cronjob was created by the operator?

Reply:
but we don't have such annotations for other resources

Reply:
Saw it in https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/resource/route.go , guess I should have checked the others also, i will remove it from mine.